core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -528,7 +528,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       }
 
     // Optionally scale number of executors dynamically based on workload. Exposed for testing.
-    val dynamicAllocationEnabled = _conf.getBoolean("spark.dynamicAllocation.enabled", false)
+    val dynamicAllocationEnabled = Utils.isDynamicAllocationEnabled(_conf)
+    if (!dynamicAllocationEnabled) {
+      logInfo(s"Dynamic Allocation and num executors both set, thus dynamic allocation disabled.")
```

Comment:
This log is not right. It will be printed anytime dynamic allocation is disabled, not just when it's disabled because of the configuration conflict.